src/core/metallicroughnesseffect.cpp

```diff
@@ -718,8 +719,25 @@ void MetallicRoughnessEffect::setToneMappingAlgorithm(MetallicRoughnessEffect::T
     emit toneMappingAlgorithmChanged(algorithm);
 }
 
+Qt3DRender::QAbstractTexture *MetallicRoughnessEffect::brdfLUT() const
+{
+    return m_brdfLUTParameter->value().value<Qt3DRender::QAbstractTexture *>();
+}
+
 void MetallicRoughnessEffect::setBrdfLUT(Qt3DRender::QAbstractTexture *brdfLUT)
 {
+    auto d = Qt3DCore::QNodePrivate::get(this);
+    auto current = this->brdfLUT();
+    if (current) {
+        d->unregisterDestructionHelper(current);
+        if (current->parent() == this)
```

Comment:
Sorry I probably wasn't clear. I was trying to say that the lines 733, 734 shouldn't be needed.

Reply:
what is the effect is the last user of the texture? It's destruction will be delayed until the effect gets deleted. Is that ok?

Reply:
deleted that code